### PR TITLE
rtl_direct: loiter hold should track altitude as best effort 

### PR DESCRIPTION
**Pull request by KonradRudin**
### Solved Problem
When during the RTL loiter down, the terrain avoidance is triggered and the vehicle is kept on an altitude higher than the indented one, it will try to sink to the indented altitude again at the end of the loiter. This is due to the loiter hold after the loiter down enforcing the altitude again.

### Solution
- The loiter hold should only track the altitude as best effort but accept any altitude error. it should only check on the loiter time.
- Also make sure that the loiter hold forces the heading if necessary. Not an issue per se but it was only implicitely set from the prior mission item

### Changelog Entry
For release notes:
```
Bugfix After RTL terrain avoidance is triggered during the loiter down, make sure to not try to sink again before proceeding to the land destination
```

### Test coverage
- SITL testing: Tested in sitl with faking an altitude difference by setting the loiter hold altitude lower than the loiter down altitude in the code

### Context
Related links, screenshot before/after, video

### CODE REVIEW

File: src/modules/navigator/rtl_direct.cpp [hunk 1]
```diff
@@ -293,6 +295,14 @@ void RtlDirect::set_rtl_item()
 			if (_param_rtl_land_delay.get() < -FLT_EPSILON) {
 				mavlink_log_info(_navigator->get_mavlink_log_pub(), "RTL: completed, loitering\t");
 				events::send(events::ID("rtl_completed_loiter"), events::Log::Info, "RTL: completed, loitering");
+
+			} else {
+				/* Set the altitude tracking to best effort but not strictly enforce it */
+				altitude_acceptance_radius = FLT_MAX;
+
+				if (_force_heading) {
```
Comment (sfuhrer) [inline src/modules/navigator/rtl_direct.cpp:303]: that was a separate issue?

File: src/modules/navigator/rtl_direct.cpp [hunk 2]
```diff
@@ -370,6 +380,7 @@ void RtlDirect::set_rtl_item()
 	} else {
 		// Convert mission item to current position setpoint and make it valid.
 		if (mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current)) {
+			pos_sp_triplet->current.alt_acceptance_radius = altitude_acceptance_radius;
```
Comment (sfuhrer) [inline src/modules/navigator/rtl_direct.cpp:383]: Remind me again - with NAN it uses the default altitude acceptance radius?

Comment (KonradRudin) [reply src/modules/navigator/rtl_direct.cpp:383]: Yes, if it is NAN it uses the default one set from the params.

File: src/modules/navigator/rtl_direct.cpp (hunk 1 above)
Comment (KonradRudin) [reply src/modules/navigator/rtl_direct.cpp:303]: Which one?

**Review by sfuhrer — APPROVED** [review]

File: src/modules/navigator/rtl_direct.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/navigator/rtl_direct.cpp:303]: forcing the heading - all good, the change makes sense.

Comment (sfuhrer) [reply src/modules/navigator/rtl_direct.cpp:303]: perfect thanks for the diligence!

**Pull request merged**